docker/tpu/tpu_example.jsonnet

```diff
@@ -0,0 +1,25 @@
+local base = import 'templates/base.libsonnet';
+local tpus = import 'templates/tpus.libsonnet';
+
+local tputests = base.BaseTest {
+  frameworkPrefix: 'pl',
+  modelName: 'tpu-tests',
+  mode: 'presubmit',
+
+  timeout: 300, # 5 minutes, in seconds.
+
+  image: std.extVar('image'),
+  imageTag: std.extVar('image-tag'),
+
+  tpuSettings+: {
+    softwareVersion: 'pytorch-nightly',
+  },
+  accelerator: tpus.v3_8,
+
+  command: [
```

Comment:
we need also run `coverage` and upload the report to https://codecov.io (which most likely need to have some credentials)
```
coverage run --source pytorch_lightning -m py.test pytorch_lightning tests benchmarks -v  --durations=25 # --flake8
codecov --token $CODECOV_TOKEN
```

Reply:
I can try to get `coverage` running but I don't know yet how to do that.

This command will run on the VM with the TPU, not on the Github Action machine. I've never used `coverage` before but I can experiment to see if we can upload the report directly from the VM (we could store tokens as Kubernetes Secrets in your Google Cloud project)